providers/file/clusters.go

```diff
@@ -0,0 +1,136 @@
+/*
+Copyright 2025 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package file
+
+import (
+	"fmt"
+	"os"
+	"path/filepath"
+
+	"k8s.io/client-go/rest"
+	"k8s.io/client-go/tools/clientcmd"
+
+	"sigs.k8s.io/controller-runtime/pkg/cluster"
+)
+
+type clusters map[string]cluster.Cluster
+
+func (p *Provider) loadClusters() (clusters, error) {
+	filepaths, err := p.collectPaths()
+	if err != nil {
+		return nil, err
+	}
+
+	kubeCtxs := map[string]*rest.Config{}
+	for _, filepath := range filepaths {
+		fileKubeCtxs, err := readFile(filepath)
+		if err != nil {
+			p.log.Error(err, "failed to read kubeconfig file", "file", filepath)
+			continue
+		}
+		for name, kubeCtx := range fileKubeCtxs {
+			if _, exists := kubeCtxs[name]; exists {
+				p.log.Error(nil, "duplicate context name found", "context", name, "file", filepath)
```

Comment:
Just for my understanding, this means that context names need to be globally unique across all kubeconfigs, right? Could we document that requirement somewhere? I'm not sure it's a 100% given.

Reply:
Correct - I was thinking of using the filepath concatenated with the context name as the cluster name to make them unique. But that could lead to very long clusterNames. 
I had just accepted it for now as the file provider is meant for dev and testing, not for productive use.
I'll give the concatenating a go and see how it turns out.